src/play/utils/Move.py

```diff
@@ -1,15 +1,111 @@
+"""Class that handles moves!
+
+Supports alternative constructors using `from_*`, so that it can be
+constructed from the sgf format, gtp, and matrix locations.
+Same goes for output, using the `to_*` functions
+"""
 
-class Move:
 
+class Move:
     def __init__(self, col=None, row=None, is_pass=False):
         self.col = col
         self.row = row
         self.is_pass = is_pass
 
-    def get_loc_as_matrix_coords(self):
+    @classmethod
+    def from_sgf(cls, string):
```

Comment:
Oh, you are using `cls` instead of `self`. Is that the convention? I saw `self` somewhere and just adopted it, PyCharm also suggests `self`.

Reply:
`cls` has to do with the `@classmethod` decorator, so `cls` actually contains the class and not an instance of it! (if that makes sense?)

Reply:
Oh yes, you are right! I am used from Java to use `this`, but in Python it makes sense to use `self` _and_ `cls`: [stackoverflow.com/a/7554899/2474159](https://stackoverflow.com/a/7554899/2474159)